### PR TITLE
Fix DetailsList columnReorderProps change not causing render

### LINKED ISSUE #6311: DetailsList columnReorderOptions not updating when state updates
### Please provide a reproduction of the bug in a codepen:
 
https://codepen.io/goodoldneon/pen/xaYJBg

### Bug Report 
 
 
- __Package version(s)__: 6.67.1
- __Browser and OS versions__: Chrome 68.0.3440.106 and macOS 10.13.6
 
#### Priorities and help requested:
 
Are you willing to submit a PR to fix? I'll try
 
Requested priority: Normal
 
Products/sites affected: N/A
 
#### Describe the issue:
 
Toggling `DetailsList` prop `columnReorderOptions` via state breaks column reordering.
 
#### Actual behavior:
 
1. On initial load, column reordering works.
1. After toggling `columnReorderOptions` off, the columns still appear reorder-able but can't actually be reordered. In other words, they have the reorder icon and can be dragged.
1. After toggling `columnReorderOptions` on again, the columns no longer have any appearance of reorder-ability.
1. Future toggles keep appearing like steps 2 and 3.
 
#### Expected behavior:
 
Toggling `columnReorderOptions` off should prevent columns from being reordered, and toggling on should enable reordering.

### PR DESCRIPTION
**Pull request by goodoldneon**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #6311
- [x] Include a change request file using `$ npm run change`

#### Description of changes

Put `columnReorderProps` into `state`, and moved update from `componentDidUpdate()` to `componentWillReceiveProps()`.

Previously, `this._columnReorderProps` was always one render "behind" because it was updated in `componentDidUpdate()`. Additionally, changes to objects bound to `this` do not trigger a render like `this.setState()` does.

#### Focus areas to test

In the [DetailsList - DragDrop](http://localhost:4322/#/examples/detailslist/DragDrop) example, reorder columns after toggling "Enable Column Reorder".

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/DetailsList/DetailsHeader.base.tsx [hunk 1]
```diff
@@ -181,7 +172,7 @@ export class DetailsHeaderBase extends BaseComponent<IDetailsHeaderBaseProps, ID
       styles,
       theme
     } = this.props;
-    const columnReorderProps = this._columnReorderProps;
+    const { columnReorderProps } = this.state;
```
Comment (KevinTCoughlin) [inline packages/office-ui-fabric-react/src/components/DetailsList/DetailsHeader.base.tsx]: Nit: you can add this to the destructuring below it as its already pulling from `this.state`

Comment (goodoldneon) [reply packages/office-ui-fabric-react/src/components/DetailsList/DetailsHeader.base.tsx]: Oops, didn't see that. Do I need to make that change and commit? Or is that something you can resolve when you merge?

Comment (micahgodbolt) [reply packages/office-ui-fabric-react/src/components/DetailsList/DetailsHeader.base.tsx]: change and push to this branch

**Comment by KevinTCoughlin** [comment]
Pushed up https://github.com/OfficeDev/office-ui-fabric-react/pull/6328/commits/442a91ea4e39eb08513019c6844f0cf867597989 -- Will merge once build passes once more. Thanks again for this fix @goodoldneon!

**Pull request merged**